lib/compress.js

```diff
@@ -323,7 +323,7 @@ merge(Compressor.prototype, {
                     if (side_effects_encountered |= lvalues_encountered) continue;
 
                     // Non-constant single use vars can only be replaced in same scope.
-                    if (ref.scope !== self) {
+                    if (ref.scope.get_defun_scope() !== self) {
```

Comment:
This change was only needed to make the tests pass, although it probably also didn't work otherwise.

I'm guessing that it has to do with the introduction of block scope in ES6.